addon/serializers/echonest.js

```diff
@@ -1,26 +1,71 @@
 import Ember from 'ember';
 import DS from 'ember-data';
 
-const { isPresent } = Ember;
+const { String: { camelize, pluralize, underscore }, isBlank, isPresent } = Ember;
 const { RESTSerializer } = DS;
 
 export default RESTSerializer.extend({
     modelKey: '',
 
+    camelizeList: [],
+    removalList: [],
+
     pluralizeKey(key) {
-        return Ember.String.pluralize(key);
+        return pluralize(key);
+    },
+
+    keyForAttribute(attr/*, method*/) {
+        return underscore(attr);
+    },
+
+    camelizeProperty(hash) {
+        const camelizeList = this.get('camelizeList');
+
+        camelizeList.forEach(function(property) {
+            const newKey = camelize(property);
+            hash[newKey] = hash[property];
+
+            delete hash[property];
+        });
+
+        return hash;
+    },
+
+    removeProperty(hash) {
+        const removalList = this.get('removalList');
+
+        removalList.forEach(function(property) {
+            delete hash[property];
+        });
+
+        return hash;
     },
 
     normalizeResponse(store, primaryModelClass, payload, id, requestType) {
         const key = this.get('modelKey');
+        const camelizeList = this.get('camelizeList');
+        const removalList = this.get('removalList');
         const pluralKey = this.pluralizeKey(key);
-        if (isPresent(key) && isPresent(payload.response) && isPresent(payload.response[pluralKey])) {
-            const items = payload.response[pluralKey].map(function (item, index) {
-                item.id = index;
-                return item;
-            });
+        if (isPresent(key) && isPresent(payload.response)) {
+            if (isPresent(payload.response[pluralKey])) {
+                const items = payload.response[pluralKey].map((item, index) => {
+                    if (isBlank(item.id)) {
+                        item.id = index;
+                    }
+                    if (isPresent(camelizeList)) {
+                      item = this.camelizeProperty(item);
+                    }
+                    if (isPresent(removalList)) {
+                      item = this.removeProperty(item);
+                    }
+
+                    return item;
+                });
+                return this._super(store, primaryModelClass, { [`echonest-${key}`]: items }, id, requestType);
+            } else if (isPresent(payload.response[key])) {
+                return this._super(store, primaryModelClass, { [`echonest-${key}`]: [ payload.response[key] ] }, id, requestType);
```

Comment:
I copied your latest parent serializer. The payload needed to be in an array so I added that part on this line.